tests/credentials_test.go

```diff
@@ -292,6 +292,73 @@ var _ = Describe("[sig-compute]Guest Access Credentials", decorators.SigCompute,
 			Eventually(matcher.ThisVMI(vmi), 240*time.Second, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceUnsupportedAgent))
 		})
 	})
+	Context("with secret and noCloud propagation", func() {
+		It("[test_id:6224]should have ssh-key under authorized keys", func() {
+			secretID := "my-pub-key"
+			userData := fmt.Sprintf(
+				"#cloud-config\npassword: %s\nchpasswd: { expire: False }\n",
+				fedoraPassword,
+			)
+			vmi := tests.NewRandomVMIWithEphemeralDiskAndNoCloudUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling), userData)
+			vmi.Namespace = util.NamespaceTestDefault
+			vmi.Spec.AccessCredentials = []v1.AccessCredential{
+				{
+					SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+						Source: v1.SSHPublicKeyAccessCredentialSource{
+							Secret: &v1.AccessCredentialSecretSource{
+								SecretName: secretID,
+							},
+						},
+						PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
+							NoCloud: &v1.NoCloudSSHPublicKeyAccessCredentialPropagation{},
+						},
+					},
+				},
+			}
+
+			key1 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"
+			key2 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"
+			key3 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"
+
+			By("Creating a secret with three ssh keys")
+			secret := kubev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretID,
+					Namespace: vmi.Namespace,
+					Labels: map[string]string{
+						util.SecretLabel: secretID,
+					},
+				},
+				Type: "Opaque",
+				Data: map[string][]byte{
+					"my-key1": []byte(key1),
+					"my-key2": []byte(key2),
+					"my-key3": []byte(key3),
+				},
+			}
+			_, err := virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), &secret, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 180)
+
+			By("Verifying all three pub ssh keys in secret are in VMI guest")
+			ExecutingBatchCmd(vmi, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "login:"},
+				&expect.BSnd{S: "fedora\n"},
+				&expect.BExp{R: "Password:"},
+				&expect.BSnd{S: fedoraPassword + "\n"},
+				&expect.BExp{R: "\\$"},
+				&expect.BSnd{S: "cat /home/fedora/.ssh/authorized_keys\n"},
+				&expect.BExp{R: "test-ssh-key1"},
+				&expect.BSnd{S: "cat /home/fedora/.ssh/authorized_keys\n"},
+				&expect.BExp{R: "test-ssh-key2"},
+				&expect.BSnd{S: "cat /home/fedora/.ssh/authorized_keys\n"},
+				&expect.BExp{R: "test-ssh-key3"},
```

Comment:
IMO you can write the test with just this. Everything else can be defined in the test setup stage, can't it ?